### ISSUE TITLE
28688182: WebKit editing infrastructure should express meaningful state changes to VoiceOver

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#16083]
#### Description

Summary:
I am the developer of a popular WebKit based blog editor, MarsEdit. Although the app is regarded as relatively accessibly by many people in the VoiceOver community, the behavior of the Rich Text editor, which uses WebKit editing, is a point of criticism.

I am open to doing what I can on my own to improve the behavior of the Rich Editor for accessibility, but it strikes me that the usability of all software employing WebKit editing would be improved if Apple built in accessibility affordances at the editing level.

A few examples where my customers get confused are the fact that editing commands such as bolding, inserting a header, or inserting a list do not cause any announced change of state to VoiceOver.

I am not an expert with regard to how the software should behave ideally for VoiceOver users, but it seems to me that as a user changes formatting through the editor, the change should be announced. Also, it would seem reasonable to me that as a user is navigating the content of rich text through WebKit, the formatting attributes of the content could be announced as well.

I am eager to make MarsEdit as accessible as possible so if you have people in either the WebKit or Accessibility teams who want to work with me on this, please get them in touch.

Steps to Reproduce:
1. Open attached sample project.
2. Build and run.
3. Enable VoiceOver
4. Click buttons to bold text, insert a header, or insert a list.

Expected Results:
Some announcement should come through VoiceOver, expressing the change of state.

Actual Results:
Nothing happens.

Version:
10.12 (16A323) 

Notes:

Configuration:

Attachments:
## 

Product Version: 10.12 (16A323)
Created: 2016-10-09T21:33:06.560540
Originated: 2016-10-09T17:32:00
Open Radar Link: http://www.openradar.me/28688182
